backend/models/sequence.js

```diff
@@ -144,16 +307,59 @@ Sequence.getList = async (account, model, branch, revision, cleanResponse = fals
 };
 
 Sequence.updateSequence = async (account, model, sequenceId, data) => {
-	if (!data || !data.name || !utils.isString(data.name) || data.name === ""  || data.name.length >= 30) {
+	const toUpdate = {};
+	const toSet = {};
+	const toUnset = {};
+
+	if (!data || !sequenceEditSchema.isValidSync(data, { strict: true })) {
 		throw responseCodes.INVALID_ARGUMENTS;
 	}
 
-	const { result } = await db.update(account, sequenceCol(model),
-		{_id: utils.stringToUUID(sequenceId)}, {$set: {name: data.name}});
-	if (result.nModified === 0) {
-		throw responseCodes.SEQUENCE_NOT_FOUND;
+	await sequenceExists(account, model, sequenceId);
+
+	if (data.name) {
+		if (!utils.isString(data.name) || data.name === "" || data.name.length >= 30) {
+			throw responseCodes.INVALID_ARGUMENTS;
+		}
+
+		toSet.name = data.name;
+	}
+
+	if (data.revId) {
```

Comment:
you shouldn' tbe able to alter revId either if it's a synchro sequence, so maybe move the custom sequence check out to wrap around this and the frames?